contracts/wveOLAS.sol

```diff
@@ -0,0 +1,170 @@
+    function totalNumPoints() external view returns (uint256 numPoints);
+
+    /// @dev Gets the supply point of a specified index.
+    /// @param idx Supply point number.
+    /// @return sPoint Supply point.
+    function mapSupplyPoints(uint256 idx) external view returns (PointVoting memory sPoint);
+
+    /// @dev Gets the number of user points.
+    /// @param account Account address.
+    /// @return userNumPoints Number of user points.
+    function getNumUserPoints(address account) external view returns (uint256 userNumPoints);
+
+    /// @dev Gets the checkpoint structure at number `idx` for `account`.
+    /// @notice The out of bound condition is treated by the default code generation check.
+    /// @param account User wallet address.
+    /// @param idx User point number.
+    /// @return uPoint The requested user point.
+    function getUserPoint(address account, uint256 idx) external view returns (PointVoting memory uPoint);
+
+    /// @dev Gets voting power at a specific block number.
+    /// @param account Account address.
+    /// @param blockNumber Block number.
+    /// @return balance Voting balance / power.
+    function getPastVotes(address account, uint256 blockNumber) external view returns (uint256 balance);
+
+    /// @dev Gets the account balance at a specific block number.
+    /// @param account Account address.
+    /// @param blockNumber Block number.
+    /// @return balance Account balance.
+    function balanceOfAt(address account, uint256 blockNumber) external view returns (uint256 balance);
+
+    /// @dev Gets total token supply at a specific block number.
+    /// @param blockNumber Block number.
+    /// @return supplyAt Supply at the specified block number.
+    function totalSupplyAt(uint256 blockNumber) external view returns (uint256 supplyAt);
+
+    /// @dev Calculates total voting power at time `ts`.
+    /// @param ts Time to get total voting power at.
+    /// @return vPower Total voting power.
+    function totalSupplyLockedAtT(uint256 ts) external view returns (uint256 vPower);
+
+    /// @dev Calculate total voting power at some point in the past.
+    /// @param blockNumber Block number to calculate the total voting power at.
+    /// @return vPower Total voting power.
+    function getPastTotalSupply(uint256 blockNumber) external view returns (uint256 vPower);
+}
+
+/// @dev Zero veOLAS address.
+error ZeroVEOLASAddress();
+
+/// @dev Provided wrong timestamp.
+/// @param minTimeStamp Minimum timestamp.
+/// @param providedTimeStamp Provided timestamp.
+error WrongTimestamp(uint256 minTimeStamp, uint256 providedTimeStamp);
+
+/// @dev Called function is implemented in a specified veOLAS contract.
+/// @param ve Original veOLAS address.
+error ImplementedIn(address ve);
+
+/// @title wveOLAS - Wrapper smart contract for view functions of veOLAS contract
+/// @author AL
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+contract wveOLAS {
```

Comment:
Functions are wrapped exactly as described in the specification doc.

Reply:
All the functions needed by the Governor contract are implemented along with others affected.